### ISSUE TITLE
[Feature]: Shadows behind items

### ISSUE THREAD

**Issue by sayyid5416** [fast4x/RiMusic#1437]
### Description

Adding shadows behind items will enhance the visual appearance of whole app.

### Suggested Solution

h

**Comment by fast4x** [comment]
Example in songs list?

**Comment by sayyid5416** [comment]
> Example

Behind:
- navigation bar
- New action menu which is being discussed in #1419 
- Up Icon which appears in songs list to scroll to top
- etc

**Comment by sayyid5416** [comment]
![image](https://github.com/fast4x/RiMusic/assets/78584556/07b8d345-e121-4419-a618-3e27e1975774)

Also behind this bar

**Comment by dAtA-TRoN** [comment]
Can you provide us an example? I'm just curious what exactly you have in mind.

**Comment by sayyid5416** [comment]
You mean a demo? wait

**Comment by dAtA-TRoN** [comment]
> You mean a demo? wait

Like this?

https://github.com/fast4x/RiMusic/assets/71184627/365665e3-d8bb-4b1f-91ed-000d07133256

**Comment by sayyid5416** [comment]
![aaa](https://github.com/fast4x/RiMusic/assets/78584556/2bb7ae2f-9ce9-4f31-b2df-6c17dfaa5bc5)

<br>

![sssssssssssss](https://github.com/fast4x/RiMusic/assets/78584556/845baeaf-a9c4-4e26-b4da-b110a1c52fbb)

**Comment by dAtA-TRoN** [comment]
Fantastic idea. This will dramatically improve the design of the app 👍🏼.

**Comment by sayyid5416** [comment]
indeed. I can do that in Qt, but I've no idea if it's possible in kotlin.

**Comment by sayyid5416** [comment]
- I already sent idea for bottom navigation above
- I'm sending **_few_** ideas for side navigation bar _(obviously they need to be refined later)_

### IDEA 1

![1](https://github.com/fast4x/RiMusic/assets/78584556/c2ee9710-e806-4120-991a-a44a064a5381)

**Comment by dAtA-TRoN** [comment]
I anticipated you would approach it that way: The bottom bar looks quite sleek, but I find the sidebar to be somewhat peculiar. Initially, my sidebar mirrored yours in length, extending across the entire screen edge. However, I began to feel it appeared somewhat awkward, emphasizing the unused space. Consequently, I chose to shorten its display, as you've seen. Furthermore, I'm of the opinion that it shouldn't remain solely white (perhaps in the light theme). It was essential to me that the bar distinctly contrasts with the other buttons, not solely through a line, as users must immediately recognize it as the main navigation.

And how do you envision this drop shadow style within the black and pitch-black theme?

**Comment by dAtA-TRoN** [comment]
> ### IDEA 2
> ![2](https://github.com/fast4x/RiMusic/assets/78584556/5be42c93-7af7-454f-b99d-289f9985067d)
> 

Oh f*ck, it actually works 😂. I imagined this whole thing would look kind of dumb. But hey, I love it!

**Comment by sayyid5416** [comment]
See.... you like it. I knew it 😏

**Comment by sayyid5416** [comment]
Regarding dark mode and pitch black mode: Drop shadow works with them too, you just need to modify the shadow color, blur, and opacity a little bit.

**Comment by dAtA-TRoN** [comment]
Please make a Mockup on this. Because we still have to sell it to fast4x 👀.

**Comment by sayyid5416** [comment]
### IDEA 2 - Dark mode
![3](https://github.com/fast4x/RiMusic/assets/78584556/d70bc49f-ecf0-4516-a74c-99d781f88898)

Shadow got too harsh (you got the idea btw)

**Comment by dAtA-TRoN** [comment]
Cool cool, but black & pitch black are the true end bosses here.

**Comment by sayyid5416** [comment]
I tried for pitch black, but I'm not master in photoshop - don't know how to create grey shadow.

A greyish shadow would work though I know it.

**Comment by sayyid5416** [comment]
> > I think the edges should be bright instead.
> 
> Not too bright though…

All these things can be fine tuned programmatically, these images are just for rough idea.

**Comment by sayyid5416** [comment]
![image](https://github.com/fast4x/RiMusic/assets/78584556/04d1fa62-0a9d-4f5c-9456-eb41e65dd1a8)

I tried to create a very big shadow, but no color is working in photoshop over pure black.

![MeBabyGIF](https://github.com/fast4x/RiMusic/assets/78584556/74612079-1035-4e8b-ac2b-6cbc83cf5383)


There would be a way but I think its not worth it to learn that now 🤷‍♂️
![YawnTomAndJerryGIF](https://github.com/fast4x/RiMusic/assets/78584556/7fbc84f8-0b88-4c19-a836-61279c036b29)

**Comment by dAtA-TRoN** [comment]
> I tried to create a very big yellow shadow _(to test)_, but no color is working in photoshop over pure black.

> There would be a way but I think its not worth it to learn that now 🤷‍♂️

Alternatively, the black and pitch-black themes simply remain flat, without a drop shadow because where there is no light, there is no shadow either 🤷🏻‍♂️.

**Comment by sayyid5416** [comment]
> > I tried to create a very big yellow shadow _(to test)_, but no color is working in photoshop over pure black.
> 
> > There would be a way but I think its not worth it to learn that now 🤷‍♂️
> 
> Alternatively, the black and pitch-black themes simply remain flat, without a drop shadow because where there is no light, there is no shadow either 🤷🏻‍♂️.

Well, That's another approach

![ThinkSmartGIF](https://github.com/fast4x/RiMusic/assets/78584556/494e932d-30b0-4495-9f4a-9df9c8484cd6)

**Comment by sayyid5416** [comment]
@fast4x I just noticed there is a shadow behind the top menu. Can that be implemented in these areas?

**Comment by fast4x** [comment]
i think not
